Automattic/woocommerce-payments
# Move the Instant deposit button into the available balance block

Fixes #4484

#### Changes proposed in this Pull Request

This PR moves the "**Instant deposit**" button from the **Payments → Overview** page block header and into the "**Available balance**" block. Inline with the proposed designs: W8FHz6thdADIqhlPlBo4Mf-fi-406%3A16975

| Before | After |
| --- | --- |
|<img width="692" alt="Screen Shot 2022-08-16 at 12 42 40 pm" src="https://user-images.githubusercontent.com/8490476/184787091-552c4538-d550-49d2-98cb-f20594a04f1a.png">| <img width="692" alt="Screen Shot 2022-08-16 at 12 41 16 pm" src="https://user-images.githubusercontent.com/8490476/184786920-3c145146-3f85-42d4-8ce8-600c8a18ea96.png">|





#### Testing instructions

Because instant deposit eligibility is locked behind [transaction volume](https://github.com/Automattic/woocommerce-payments-server/blob/trunk/server/wp-content/rest-api-plugins/endpoints/wcpay/services/class-deposit-service.php#L370-L373) and other criteria, I had to hack my local wcpay server for the Instant deposit button to show up. The changes I made to server are:
- `Deposit_Service::calculate_instant_deposit_eligibility()` must return `true`.
- `Deposit_Service::get_instant_balances_available()` needs to return a balance that is eligible for instant deposit. For my account, I had to bypass [this condition](https://github.com/Automattic/woocommerce-payments-server/blob/trunk/server/wp-content/rest-api-plugins/endpoints/wcpay/services/class-deposit-service.php#L554). 


<details>
  <summary>Example Diff</summary>

  ```diff
diff --git a/server/wp-content/rest-api-plugins/endpoints/wcpay/services/class-deposit-service.php b/server/wp-content/rest-api-plugins/endpoints/wcpay/services/class-deposit-service.php
index 1e7903b99..efd7bb59d 100644
--- a/server/wp-content/rest-api-plugins/endpoints/wcpay/services/class-deposit-service.php
+++ b/server/wp-content/rest-api-plugins/endpoints/wcpay/services/class-deposit-service.php
@@ -347,6 +347,7 @@ class Deposit_Service extends Service {
         * @throws \WCPay\Core\Exceptions\Server_Exception
         */
        public function calculate_instant_deposit_eligibility( Account $account ): bool {
+               return true;
                // Limit to US accounts for now, until we work out the kinks with multi-currency deposits.
                if ( Country::US !== $account->country ) {
                        return false;
@@ -549,9 +550,10 @@ class Deposit_Service extends Service {
        public function get_instant_balances_available( array $balance, array $stripe_account, bool $test_mode ): array {
                // Identify external accounts with available instant deposits.
                $deposit_accounts = [];
+
                // external_accounts is not set for live accounts when using test_mode=true.
                foreach ( $stripe_account['external_accounts']['data'] ?? [] as $deposit_account ) {
-                       if ( in_array( 'instant', $deposit_account['available_payout_methods'], true ) ) {
+                       if ( true || in_array( 'instant', $deposit_account['available_payout_methods'], true ) ) {
                                $deposit_accounts [] = $deposit_account;
                        }
                }
(END)
  ```
</details>

1. Make sure your store is eligible for instant deposit, with the changes above to make your account eligible. 
2. Go to **Payments → Overview**
    - On `develop`, the "Instant deposit" button will appear in the top right. 
    - On this branch the  "Instant deposit" button will appear in the "Available balance" block. 
3. Clicking on the button should still result in the Instant deposit modal being shown (see screenshot below). 

<p align="center">
  <img width="581" alt="Screen Shot 2022-08-16 at 3 21 57 pm" src="https://user-images.githubusercontent.com/8490476/184803749-9c915092-1dd9-4aaf-b24a-32f308ff2605.png">
</p>


-------------------


**Post merge**




**Bonus post release task**
- [ ] Update the screenshots and documentation about this feature here: https://woocommerce.com/document/payments/instant-deposits/
